upgrade50Plus.sh

```diff
@@ -53,18 +87,122 @@ print_dd_progress() {
+    executeOnWin "exit" || quit "Can't login into Windows" "Update failed." "" "Update failed." "No Windows login"
+    pretty "Checking connection..." "done." "" "Checking connection..." "done."
+}
+
+get_hdw_info() {
+    # complete diskdrive info via 'wmic diskdrive where (DeviceID='\\\\.\\PHYSICALDRIVE0') list /format:list'
+    # EXAMPLE: 32 GB SSD
+    #
+    # BytesPerSector=512                    Capabilities={3,4,10}               CapabilityDescriptions={"Random Access","Supports Writing","SMART Notification"}
+    # ConfigManagerErrorCode=0              ConfigManagerUserConfig=FALSE       Description=Disk drive
+    # DeviceID=\\.\PHYSICALDRIVE0           Index=0                             InterfaceType=IDE
+    # Manufacturer=(Standard disk drives)   MediaLoaded=TRUE                    MediaType=Fixed hard disk media
+    # Model=TS32GMTS800S                    Name=\\.\PHYSICALDRIVE0             Partitions=4
+    # PNPDeviceID=SCSI\DISK&amp;VEN_&amp;PROD_TS32GMTS800S\4&amp;20157ED9&amp;0&amp;020000
+    # SCSIBus=2                             SCSILogicalUnit=0                   SCSIPort=0
+    # SCSITargetId=0                        SectorsPerTrack=63                  Size=32012789760
+    # Status=OK                             SystemName=DESKTOP-O4RBF7E          TotalCylinders=3892
+    # TotalHeads=255                        TotalSectors=62524980               TotalTracks=992460
+    # TracksPerCylinder=255
+
+    pretty "Getting hardware info..." "" "" "Getting hardware info..." ""
+    SSD_SIZE=$($SSHPASS_PATH/sshpass -p 'TEST' ssh -o StrictHostKeyChecking=no TEST@$IP "wmic diskdrive where (DeviceID='\\\\\\\\.\\\\PHYSICALDRIVE0') get size")
+    SSD_SIZE=$(echo "$SSD_SIZE" | sed -n 2p)        # Size Info is in the second line
+    SSD_SIZE=${SSD_SIZE//[ $'\001'-$'\037']}        # remove possible DOS carriage return characters
+    SSD_SIZE=$((SSD_SIZE / 1000000000))
+    if ! [[ $SSD_SIZE =~ ^[0-9]+$ ]]; then
+        quit "SSD size is NaN" "" "" "SSD size" "is NaN"
+    fi
+    pretty "Getting hardware info..." "done." "" "Getting hardware info..." "done."
+}
+
+check_preconditions_win() {
+    pretty "Checking preconditions..." "" "" "Checking" "preconditions..."
+    if [ $SSD_SIZE -eq 32 ]; then
+        pretty "ePC SSD Size" "$SSD_SIZE GB" "" "ePC SSD Size" "$SSD_SIZE GB"
+        executeOnWin "wmic partition get StartingOffset | findstr "21529362432"" || \
+            quit "Unexpected partition" "ePC partition 3 is not at" "expected position, update failed." "Update failed." "partition error"
+        executeOnWin "wmic partition get NumberOfBlocks | findstr "20482422"" || \
+            quit "Unexpected partition" "ePC partition 3 is not of" "expected size, update failed." "Update failed." "partition error"
+        PART_POS=$(( 21529362432 / 512))  # 42049536 Sectors
+        PART_SIZE=20482422
+    elif [ $SSD_SIZE -eq 64 ]; then
+        pretty "ePC SSD Size" "$SSD_SIZE GB" "" "ePC SSD Size" "$SSD_SIZE GB"
```

Comment:
This looks wuite copy 'n pasted. Please extract function that takes the size as argument and call it in each if/else branch.